core/repository/sail/src/test/java/org/eclipse/rdf4j/repository/sail/SailRepositoryConnectionTest.java

```diff
@@ -148,4 +152,21 @@ public void testPrepareBooleanQuery_bypassed() {
 		verify(sailConnection).evaluate(eq(expr), any(), any(), anyBoolean());
 	}
 
+	@Test
+	public void testExplainQuery() {
+		TupleExpr expr = mock(TupleExpr.class);
+		when(expr.clone()).thenReturn(expr);
+		Explanation explanation = mock(Explanation.class);
+
+		when(sailConnection.prepareQuery(any(), any(), any(), any())).thenReturn(Optional.of(expr));
+		when(sailConnection.explain(any(), any(TupleExpr.class), any(), any(), anyBoolean(), anyInt()))
+				.thenReturn(explanation);
+
+		TupleQuery query = subject.prepareTupleQuery("SELECT * WHERE { ?s ?p ?o }");
+		assertThat(query.explain(Explanation.Level.Unoptimized)).isEqualTo(explanation);
+
+		verify(sailConnection).explain(eq(Explanation.Level.Unoptimized), any(QueryRoot.class), any(), any(),
```

Comment:
And here I wanted to use
```java
verify(sailConnection).explain(eq(Explanation.Level.Unoptimized), eq(new QueryRoot(expr)), any(), any(),
				anyBoolean(), anyInt());
```
but that didn't work.

Reply:
Should be fine as it is.